realm/realm-library/src/androidTestObjectServer/java/io/realm/ObjectLevelPermissionsTest.java

```diff
@@ -430,6 +432,20 @@ public void getClassPermissions_closedRealmThrows() {
 //        }
     }
 
+    @Test
+    public void userPrivateRole() {
+        RealmResults<PermissionUser> permissionUsers = realm.where(PermissionUser.class).findAll();
+        assertEquals(1, permissionUsers.size());
+
+        PermissionUser permissionUser = permissionUsers.get(0);
```

Comment:
It isn't clear to me where this user is created?

Reply:
Perhaps also create a test for what happens if `createObject(PermissionUser.class)` is called as I suspect that the `role` isn't created until you talked to the server?

E.g. what happens if you do 
```
Role role = getExistingRole();
role.addMember(newUserId)`
realm.where(PermissionUser.class).equalTo("id", newUserId).findFirst().getRole(); // What is returned here?
```

Reply:
@cmelchior user is created during the `setUp` (`user = createTestUser();`)